tools/bridge/bridge/event_fetcher.py

```diff
@@ -48,14 +49,42 @@ def __init__(
         self.event_queue = event_queue
         self.max_reorg_depth = max_reorg_depth
         self.last_fetched_block_number = start_block_number - 1
+        self._setup_tenacity()
+
+    def _setup_tenacity(self):
+        # we do not use the decorator syntax, because we do want to
+        # bind this to our instance local logger.
```

Comment:
I think without the decorator syntax this could be really confusing. I wouldn't look in `__init__` for anything overwriting a method. Could we either:

- rename the `_rpc_*` functions to something like `_try_rpc_*_once`, so that the `retry` method has at least a different name
- implement a custom `retry` decorator wrapped around the tenacity one, but uses the local logger

?

Reply:
I think your first suggestion makes it even worse, since the reader will now even think that the method is missing.
I will look into the second option.

Reply:
I've updated the code to explicitly use tenacity's Retrying object. Please have a look at it again.